elements/lisk-tree/src/merkle_tree/calculate.ts

```diff
@@ -0,0 +1,153 @@
+import { hash } from "../../../lisk-cryptography/dist-node";
+import { LEAF_PREFIX, NODE_HASH_SIZE } from "./constants";
+import { NodeIndex, NodeInfo, NodeLocation, NodeSide, NonNullableStruct, Proof, SiblingHash } from "./types";
+import { buildBranch, createNewBranchNode, createNewLeafNode, generateNode, getBinaryString,
+    getHeight, getPairLocation, getParentLocation, isLocationEqualToSiblingHash } from "./utils";
+
+export const calculatePathNodes = (
+    queryData: ReadonlyArray<Buffer>,
+    size: number,
+    idxs: ReadonlyArray<NodeIndex>,
+    siblingHashes: ReadonlyArray<SiblingHash>) => {
```

Comment:
type of idxs and siblingHashes should be
```
idxs: ReadonlyArray<number>,
siblingHashes: ReadonlyArray<Buffer>,
```
otherwise input of https://github.com/LiskHQ/lisk-sdk/pull/6585/files#diff-ab59b447c02797885741efe9ce43a0320e12f69ab1675e638f825b52637efe57R149 will be invalid
also, it should follow https://github.com/LiskHQ/lips-staging/blob/master/proposals/lip-0031.md#proof-serialization

Reply:
We might need to change the `number` to `bigint` later, but this will apply everywhere else, so let's not change it now

Reply:
done